src/plugins/intel_gpu/src/graph/primitive_inst.cpp

```diff
@@ -127,8 +127,18 @@ void primitive_inst::realloc_if_needed() {
 
 void primitive_inst::update_impl() {
     if (!_node.is_type<data>() && !(_node.is_type<mutable_data>() && _node.get_dependencies().empty())) {
-        _impl = std::move(_node.type()->choose_impl(_node));
-        _network.get_program()->compile();
+        auto layout_key = _node.get_output_layout().to_string();
+        auto caches = _node.get_primitive_impl_caches();
```

Comment:
There is one "cache" for a primitive. So "caches" seems to be misread.

Reply:
Fixed it.